lib/info.dart

```diff
@@ -46,8 +50,26 @@ abstract class Info {
 // TODO(sigmund): add more:
 //  - inputSize: bytes used in the Dart source program
 abstract class BasicInfo implements Info {
+  static Set<int> _ids = new Set<int>();
   final InfoKind kind;
-  final int id;
+
+  int _id;
+  int get id {
+    if (_id == null) {
+      // we hope we have all of the parents set up by now...
```

Comment:
Grr...this gets ugly quickly. We have to catch all of the access to these guys ahead of time. Don't think it's worth it...